src/tools/idyntree-model-simplify-shapes.cpp

```diff
@@ -0,0 +1,83 @@
+/*
+ * SPDX-FileCopyrightText: 2006-2021 Istituto Italiano di Tecnologia (IIT)
+ * SPDX-License-Identifier: LGPL-2.1-or-later
+ */
+
+#include <iDynTree/Model/Model.h>
+
+// To load models from file
+#include <iDynTree/ModelIO/ModelLoader.h>
+
+// To approximate shapes of a model
+#include <iDynTree/ModelTransformersSolidShapes.h>
+
+// To write model to file
+#include <iDynTree/ModelIO/ModelExporter.h>
+
+#include "cmdline.h"
+
+#include <iostream>
+#include <cstdlib>
+
+/**
+ * Add the option supported by the idyntree-model-simplify-shapes utility
+ */
+void addOptions(cmdline::parser &cmd)
+{
+    cmd.add<std::string>("model", 'm',
+                         "Input model to load and simplify.",
+                         true);
+
+    cmd.add<std::string>("output-model", 'o',
+                         "Output simplified model.",
+                         true);
+}
+
+int main(int argc, char** argv)
+{
+    cmdline::parser cmd;
+    addOptions(cmd);
+    cmd.parse_check(argc, argv);
+
+    // Read model
+    const std::string& modelPath = cmd.get<std::string>("model");
+    const std::string& outputModelPath = cmd.get<std::string>("output-model");
```

Comment:
Maybe it is worth adding a message and exit prematurely in case the tool is not used correctly.

Reply:
If I remember correctly this is handled automatically by the `cmdline` library, let me check.

Reply:
Yes, whenever the option are not passed the help message is printed again.